doc/source/train/user-guides/persistent-storage.rst

```diff
@@ -1,62 +1,387 @@
+cloud storage (e.g., S3, GCS) or NFS (e.g., AWS EFS, Google Filestore) is
+highly recommended when doing multi-node training.
+
+Here are some benefits of setting up persistent storage:
+
+- **Checkpointing and fault tolerance**: Saving checkpoints to a persistent storage location
+  allows you to resume training from the last checkpoint in case of a node failure.
+  See :ref:`train-checkpointing` for a detailed guide on how to set up checkpointing.
+- **Post-experiment analysis**: A consolidated location storing data from all trials is useful for post-experiment analysis
+  such as accessing the best checkpoints and hyperparameter configs after the cluster has already been terminated.
+- **Bridge with downstream serving/batch inference tasks**: You can easily access the models
+  and artifacts to share them with others or use them in downstream tasks.
+
+
+Cloud storage (AWS S3, Google Cloud Storage)
+--------------------------------------------
+
+.. tip::
+
+    Cloud storage is the recommended persistent storage option.
+
+If all nodes in a Ray cluster have access to cloud storage, then all outputs can be uploaded to a shared cloud bucket.
+
+Use cloud storage by specifying a bucket URI as the :class:`RunConfig(storage_path) <ray.train.RunConfig>`:
+
+.. code-block:: python
+
+    from ray import train
+    from ray.train.torch import TorchTrainer
+
+    trainer = TorchTrainer(
+        ...,
+        run_config=train.RunConfig(
+            storage_path="s3://bucket-name/sub-path/",
+            name="experiment_name",
+        )
+    )
+
+
+In this example, all files are uploaded to shared storage at ``s3://bucket-name/sub-path/experiment_name`` for further processing.
+
+
+Network filesystem (NFS)
+------------------------
+
+If all Ray nodes have access to a network filesystem, e.g. AWS EFS or Google Cloud Filestore,
+then all outputs can be saved to this shared filesystem.
+
+Use NFS by specifying the mount path as the :class:`RunConfig(storage_path) <ray.train.RunConfig>`:
+
+.. code-block:: python
+
+    from ray import train
+    from ray.train.torch import TorchTrainer
+
+    trainer = TorchTrainer(
+        ...,
+        run_config=train.RunConfig(
+            storage_path="/mnt/cluster_storage",
+            name="experiment_name",
+        )
+    )
+
+In this example, all files are saved to ``/mnt/cluster_storage/experiment_name`` for further processing.
+
+
+Local storage
+-------------
+
+Using local storage for a single-node cluster
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you're just running an experiment on a single node (e.g., on a laptop), Ray Train will use the
+local filesystem as the storage location for checkpoints and other artifacts.
+Results are saved to ``~/ray_results`` in a sub-directory with a unique auto-generated name by default,
+unless you customize this with ``storage_path`` and ``name`` in :class:`~ray.train.RunConfig`.
```

Comment:
What if the users specify `storage_path="customized local directory"`?

Reply:
yep shown in the example in this section